Src/coffeescript/cql-execution/src/elm/interval.coffee

```diff
@@ -133,4 +133,36 @@ module.exports.Starts = class Starts extends UnimplementedExpression
 
 module.exports.Ends = class Ends extends UnimplementedExpression
 
-module.exports.Collapse = class Collapse extends UnimplementedExpression
+module.exports.Collapse = class Collapse extends Expression
+  constructor: (json) ->
+    super
+
+  exec: (ctx) ->
+    result = @execArgs ctx
+    if result?.length > 1
+      # we don't handle imprecise intervals at this time
+      for a in result
+        if a.low.isImprecise?() || a.high.isImprecise?()
+          throw new Error("Collapse does not support imprecise dates at this time.")
+
+      # sort intervals by start
+      result.sort (a,b)->
+        return -1 if a.low < b.low
+        return 1 if a.low > b.low
```

Comment:
If `a.low` is an instance of `DateTime`, I don't think you can use native `<` and `>` here.  You probably have to do something like:
```
if typeof a.low.before == 'function'
  return -1 if a.low.before b.low
  return 1 if a.low.after b.low
else
  return -1 if a.low < b.low
  return 1 if a.low > b.low
0
```